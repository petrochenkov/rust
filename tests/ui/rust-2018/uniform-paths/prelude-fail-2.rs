//@ edition:2018

// Built-in attribute
use inline as imported_inline;
mod builtin {
    pub use inline as imported_inline;
}

// Tool module
use rustfmt as imported_rustfmt;
mod tool_mod {
    pub use rustfmt as imported_rustfmt;
}

#[imported_inline]
#[builtin::imported_inline] //~ ERROR cannot determine resolution for the attribute macro
#[imported_rustfmt::skip] //~ ERROR cannot use a tool module through an import
#[tool_mod::imported_rustfmt::skip] //~ ERROR cannot use a tool module through an import
fn main() {}
