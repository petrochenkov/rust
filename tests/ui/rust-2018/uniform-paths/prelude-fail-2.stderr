error: cannot use a tool module through an import
  --> $DIR/prelude-fail-2.rs:17:3
   |
LL | #[imported_rustfmt::skip]
   |   ^^^^^^^^^^^^^^^^
   |
note: the tool module imported here
  --> $DIR/prelude-fail-2.rs:10:5
   |
LL | use rustfmt as imported_rustfmt;
   |     ^^^^^^^^^^^^^^^^^^^^^^^^^^^

error: cannot use a tool module through an import
  --> $DIR/prelude-fail-2.rs:18:13
   |
LL | #[tool_mod::imported_rustfmt::skip]
   |             ^^^^^^^^^^^^^^^^
   |
note: the tool module imported here
  --> $DIR/prelude-fail-2.rs:12:13
   |
LL |     pub use rustfmt as imported_rustfmt;
   |             ^^^^^^^^^^^^^^^^^^^^^^^^^^^

error: cannot determine resolution for the attribute macro `builtin::imported_inline`
  --> $DIR/prelude-fail-2.rs:16:3
   |
LL | #[builtin::imported_inline]
   |   ^^^^^^^^^^^^^^^^^^^^^^^^
   |
   = note: import resolution is stuck, try simplifying macro imports

error: aborting due to 3 previous errors

